core-api/src/main/java/com/optimizely/ab/config/audience/UserAttribute.java

```diff
@@ -45,12 +45,12 @@ public String getType() {
         return type;
     }
 
-    public String getValue() {
+    public Object getValue() {
         return value;
     }
 
-    public boolean evaluate(Map<String, String> attributes) {
-        String userAttributeValue = attributes.get(name);
+    public boolean evaluate(Map<String, ?> attributes) {
+        Object userAttributeValue = attributes.get(name);
```

Comment:
Hmm should be careful with this. Eventually when we do accept objects and arrays to be passed in, the `equals` evaluation below will likely perform a reference check as well. That said, I think it's fine for now, but just leave a message saying that we should be extra considerate when we decide to allow objects and arrays.